graql/reasoner/atom/task/relate/RelationSemanticProcessor.java

```diff
@@ -155,11 +156,11 @@ public MultiUnifier getMultiUnifier(RelationAtom childAtom, Atom parentAtom, Uni
                     Set<Type> parentTypes = parentVarTypeMap.get(parentRolePlayer);
                     Type parentTypeExact = parentAtom.getPredicates(parentRolePlayer, IdPredicate.class)
                             .filter(idPredicate ->  !idPredicate.isPlaceholder())
-                            .findAny()
                             .map(idPredicate -> (Concept)ctx.conceptManager().getConcept(idPredicate.getPredicate()))
+                            .filter(Objects::nonNull)
```

Comment:
how can we have null IDs?

Reply:
ids are not null but a lookup can return null if it doesn't exist